src/optimizer/query_planner.c

```diff
@@ -6743,9 +6744,14 @@ planner_visit_node (QO_PLANNER * planner, QO_PARTITION * partition, PT_HINT_ENUM
 	    edge_cnt++;
 
 	    /* set join type */
-	    if (join_type == NO_JOIN)
+	    if (join_type == NO_JOIN || is_dummy_term)
```

Comment:
I thank it needs a comment to replace as-is comment explains why **is_dummy_term** is applied.

Reply:
Okay. I'll fix it